storage/azure/src/integration-test/java/io/aiven/kafka/tieredstorage/storage/azure/AzureBlobStorageMetricsTest.java

```diff
@@ -0,0 +1,185 @@
+            .replace(")", "")
+            .replace("[", "")
+            .replace("]", "");
+        while (azureContainerName.length() < 3) {
+            azureContainerName += azureContainerName;
+        }
+        blobServiceClient.createBlobContainer(azureContainerName);
+    }
+
+    StorageBackend storage() {
+        final AzureBlobStorage azureBlobStorage = new AzureBlobStorage();
+        // The well-known Azurite account name and key.
+        final String accountName = "devstoreaccount1";
+        final String accountKey =
+            "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==";
+        final Map<String, Object> configs = Map.of(
+            "azure.container.name", azureContainerName,
+            "azure.account.name", accountName,
+            "azure.account.key", accountKey,
+            "azure.endpoint.url", endpoint(AZURITE_SERVER, BLOB_STORAGE_PORT),
+            "azure.upload.block.size", UPLOAD_BLOCK_SIZE
+        );
+        azureBlobStorage.configure(configs);
+        return azureBlobStorage;
+    }
+
+    static Stream<Arguments> metricsShouldBeReported() {
+        return Stream.of(
+            Arguments.of(
+                Named.of("smaller-than-block-size-payload", UPLOAD_BLOCK_SIZE - 1),
+                1, 0, 0),
+            Arguments.of(
+                Named.of("larger-than-block-size-payload", UPLOAD_BLOCK_SIZE + 1),
+                0, 2, 1)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void metricsShouldBeReported(
+        final int uploadBlockSize,
+        final double expectedPutBlob,
+        final double expectedPutBlock,
+        final double expectedPutBlockList
+    ) throws StorageBackendException, IOException, JMException {
+        final byte[] data = new byte[uploadBlockSize];
+
+        final ObjectKey key = new TestObjectKey("x");
+
+        final var storage = storage();
+
+        storage.upload(new ByteArrayInputStream(data), key);
+        try (final InputStream fetch = storage.fetch(key)) {
+            fetch.readAllBytes();
+        }
+        try (final InputStream fetch = storage.fetch(key, BytesRange.of(0, 1))) {
+            fetch.readAllBytes();
+        }
+        storage.delete(key);
+
+        final ObjectName objectName =
+            ObjectName.getInstance("aiven.kafka.server.tieredstorage.azure:type=azure-blob-metrics");
+        assertThat(MBEAN_SERVER.getAttribute(objectName, "blob-get-rate"))
+            .asInstanceOf(DOUBLE)
+            .isGreaterThan(0.0);
+        assertThat(MBEAN_SERVER.getAttribute(objectName, "blob-get-total"))
+            .isEqualTo(2.0);
+
+        if (expectedPutBlob > 0) {
+            assertThat(MBEAN_SERVER.getAttribute(objectName, "blob-upload-rate"))
+                .asInstanceOf(DOUBLE)
+                .isGreaterThan(0.0);
+            assertThat(MBEAN_SERVER.getAttribute(objectName, "blob-upload-total"))
+                .isEqualTo(expectedPutBlob);
```

Comment:
I guess this should be outside of the `if` (and others below, too)? We want to check it always.